Dockerfile

```diff
@@ -1,21 +1,45 @@
-FROM quay.io/centos/centos:stream8
+# build poetry
+FROM quay.io/centos/centos:stream8 as poetry
 
 RUN dnf -y module install python39 && dnf -y install python39 python39-pip
-RUN mkdir /app
-ADD https://raw.githubusercontent.com/arcalot/arcaflow-plugin-template-python/main/LICENSE /app
-ADD example_plugin.py /app
-ADD test_example_plugin.py /app
-ADD requirements.txt /app
+
 WORKDIR /app
 
-RUN pip3 install -r requirements.txt
+COPY poetry.lock /app/
+COPY pyproject.toml /app/
+
+RUN python3.9 -m pip install poetry \
+ && python3.9 -m poetry config virtualenvs.create false \
+ && python3.9 -m poetry install --without dev \
+ && python3.9 -m poetry export -f requirements.txt --output requirements.txt --without-hashes
+
 
-RUN mkdir /htmlcov
-RUN pip3 install coverage
-RUN python3 -m coverage run test_example_plugin.py
-RUN python3 -m coverage html -d /htmlcov --omit=/usr/local/*
+# coverage test
+FROM poetry as coverage
 
-VOLUME /config
+COPY example_plugin.py /app/
+COPY test_example_plugin.py /app/
+
+RUN mkdir /htmlcov \
+ && python3.9 -m pip install coverage -r requirements.txt \
+ && python3.9 -m coverage run test_example_plugin.py \
+ && python3.9 -m coverage html -d /htmlcov --omit=/usr/local/*
+
+
+# final image
+FROM quay.io/centos/centos:stream8
+
+RUN dnf -y module install python39 && dnf -y install python39 python39-pip
+
+WORKDIR /app
+
+COPY --from=poetry /app/requirements.txt /app/
+COPY --from=coverage /htmlcov /htmlcov/
```

Comment:
Why does the final image need to contain the coverage HTML report?

Reply:
Why are we not building the base image, then after building it, running the coverage in a separate image built on top of the base image to verify it?

Reply:
@mfleader Has this changed at all with the last big carpenter update? Is this still necessary in the plugin Dockerfile to have `/htmlcov` in the final image?

Reply:
I have removed htmlcov from final image after checking with @mfleader and refactored the testing in the first stage.